### PR TITLE
Allow user to customize versions

### PR DESCRIPTION
**Pull request by ahmedalsudani**

### CODE REVIEW

**Comment by ahmedalsudani** [comment]
This PR has revealed a problem to me. Some of the expectations in `version.spec.js` aren't being evaluated. I tried to figure out what exactly the problem is, but I've had no luck.

Going to look into it again over the next couple of days

File: tests/version.spec.js [hunk 1]
```diff
@@ -124,16 +124,27 @@ describe('Version', function() {
           JSON.stringify(mockVersions));
         });
 
-      version.fetchAllVersions('/versions', function(versions) {
-        expect(versions).to.be.equal(mockVersions.list);
```
Comment (ahmedalsudani) [inline tests/version.spec.js]: This is the expectation that was not actually firing. It's not firing after the changes made in this PR either.

**Comment by winkerVSbecks** [comment]
@ahmedalsudani are you still working on this PR or is it good to review?

**Comment by ahmedalsudani** [comment]
@winkerVSbecks the code is good to go, but there's a problem with the tests. I'll try to fix it this morning, but if it's too tricky, I'll save it for the next PR.

BTW, I was asking about a different PR yesterday that I haven't submitted yet. I forgot about this one :smile:

**Comment by ahmedalsudani** [comment]
@winkerVSbecks I have not figured out why that test is failing. I'll look at it in the next PR.

The situation is:

- the test did not work in the past
- the test does not work now
- the code seems to work (It's being used by 2 projects)

I think we should merge it if the code passes review.

**Comment by winkerVSbecks** [comment]
@ahmedalsudani ok sounds good. :shipit:

**Pull request merged**